README.md

```diff
@@ -49,3 +49,56 @@ tree.Sorter = func(u1, u2 fyne.URI) bool {
 <p align="center" markdown="1" style="max-width: 100%">
   <img src="img/widget-filetree.png" width="1024" height="880" alt="FileTree Widget" style="max-width: 100%" />
 </p>
+
+### CompletionEntry
+
+An extension of widget.Entry for displaying a popup menu for completion. The "up" and "down" keys on the keyboard are used to navigate through the menu, the "Enter" key is used to confirm the selection. The options can also be selected with the mouse. The "Escape" key closes the selection list.
+
+```go
+entry := widget.NewCompletionEntry([]string{})
+
+// When the use typed text, complete the list.
+entry.OnChanged = func(s string) {
+    // completion start for text length >= 3
+    if len(s) < 3 {
+        entry.HideCompletion()
+        return
+    }
+
+    // Make a search on wikipedia
+    resp, err := http.Get(
+        fmt.Sprintf(
+            "https://en.wikipedia.org/w/api.php?action=opensearch&search=%s", entry.Text,
+        ),
+    )
+    if err != nil {
+        entry.HideCompletion()
```

Comment:
If you return here like above then the remaining code can avoid being inside an else.
We generally prefer this so that the "main path" is top level in a method.

Reply:
Ha, that was because I made some other tests and forget to remove the block.

Reply:
Pretty sure this is missing a return?